job/deployer.go

```diff
@@ -120,14 +147,104 @@ func (d *deployer) cleanPerNamespace(ctx context.Context, namespaceSpec models.N
 	return nil
 }
 
-func (d *deployer) enrichJobSpecWithHookDependencies(jobSpecs []models.JobSpec) []models.JobSpec {
-	var enrichedJobSpecs []models.JobSpec
-	for _, jobSpec := range jobSpecs {
-		hooks := d.dependencyResolver.FetchHookWithDependencies(jobSpec)
-		if len(hooks) > 0 {
-			jobSpec.Hooks = hooks
+func (d *deployer) enrichJobSpecs(ctx context.Context, jobSpecs []models.JobSpec, deploymentProjectSpec models.ProjectSpec) error {
+	jobsByDestination := models.JobSpecs(jobSpecs).GroupJobsByDestination()
```

Comment:
jobsByDestination is only being used inside `getJobIDResourceDependenciesMap`, then just form it inside `getJobIDResourceDependenciesMap`.